common/src/main/java/io/netty/util/concurrent/FastThreadLocal.java

```diff
@@ -57,21 +58,25 @@ public static void removeAll() {
         }
 
         try {
-            Object v = threadLocalMap.indexedVariable(variablesToRemoveIndex);
-            if (v != null && v != InternalThreadLocalMap.UNSET) {
-                @SuppressWarnings("unchecked")
-                Set<FastThreadLocal<?>> variablesToRemove = (Set<FastThreadLocal<?>>) v;
-                FastThreadLocal<?>[] variablesToRemoveArray =
-                        variablesToRemove.toArray(new FastThreadLocal[variablesToRemove.size()]);
-                for (FastThreadLocal<?> tlv: variablesToRemoveArray) {
-                    tlv.remove(threadLocalMap);
-                }
-            }
+            removeAll(threadLocalMap);
         } finally {
             InternalThreadLocalMap.remove();
         }
     }
 
+    private static void removeAll(InternalThreadLocalMap threadLocalMap) {
```

Comment:
Is there a reason this was split out into its own function?

I had expected the `Runnable` passed to `ThreadCleaner` would use it, but it uses `remove(...)` instead. It's a bit hard to determine what's the difference between the two methods.

Reply:
Actually sorry it should use `removeAll` and I messed up :(

Reply:
Sorry what I wanted to say was that we not need to factor out removeAll and can just call remove... will fix it.